### PR TITLE
Implement Project and project-relative resolving

### PR DESCRIPTION
**Pull request by oberien**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -79,31 +79,56 @@ fn main() {
...
+            }
+        } else {
+            None
+        }
     };
 
+    let file = match cfgfile_folder.as_ref() {
+        Some(folder) => {
+            let file = folder.join("heradoc.toml");
+            let content = fs::read_to_string(&file).expect(&format!("error reading config file at {:?}",file));
```
Comment (197g) [inline src/main.rs:106]: A panic on user input?

File: src/main.rs [hunk 2]
```diff
@@ -79,31 +79,56 @@ fn main() {
...
-        let content = fs::read_to_string(cfgfile).expect("error reading existing config file");
-        toml::from_str(&content).expect("invalid config")
-    } else {
-        FileConfig::default()
+    // try to find a heradoc.toml
+    // 1. passed via commandline
+    // 2. next to the input file
+    // 3. somewhere in the parent folders of the input file
+    let cfgfile_folder = match args.configfile.as_ref() {
+        Some(file) => Some(file.parent().expect("configfile passed as argument should point to a file and not the root-folder").to_owned()),
```
Comment (197g) [inline src/main.rs:87]: I see we have enough of these panic-unwraps above as well. Oh well, it might be time for a proper error with variants.

**Review by 197g — COMMENTED** [review]
Resolution and canonicalization seem fine but we should really, really switch to an error for main instead of panicking. Tell me if you're fine with doing it as part of this PR or a separate one.

**Comment by oberien** [comment]
Thanks for the review! Currently, we always panic on user input regarding command line arguments. I created #138 to track moving to proper diagnostics during argument usage and config parsing / migration. However, that depends on #137, which will be quite a huge refactor. I don't want to block this PR on all of that mountain of work, especially because currently we just panic everywhere before we call into the generator.

**Review by 197g — APPROVED** [review]

**Pull request merged**